### ISSUE TITLE
Android versions

### ISSUE THREAD

**Issue by Lanchon** [heagoo/odex_patcher#2]
which android versions does this work on?

ever since 5.0 android does ahead of time compilation to native code. how does patching an odex help?

thanks!

**Comment by giacomoferretti** [comment]
You are correct, since 5.0 they made ART the default runtime (on 4.4 was experimental opt-in). You can modify the .odex file to run arbitrary code while maintaining the original application signature.

Based on the code, this version only works from Android 6.0 to 7.1, as on Android 8.0 they added .vdex files and they need to be patched too. Also this doesn't work on x86 and x86_64 devices.

You can basically run whatever you want with the original signature.

If you'd like to try something supporting from 4.4 to 13 check this out: https://github.com/giacomoferretti/odex-patcher (Yes, it's mine)
